src/components/ebay-listbox-button/component.js

```diff
@@ -12,14 +12,20 @@ export default {
     },
 
     handleListboxChange(event) {
-        if (event.wasClicked) {
-            this._expander.expanded = false;
-        }
         const selectedIndex = event.index;
         this.state.selectedIndex = selectedIndex;
         this.emit('change', event);
     },
 
+    handleListboxSelect(event) {
+        if (this.input.collapseOnSelect) {
+            this._expander.expanded = false;
+            if (!event.wasClicked) this.el.firstChild.focus();
```

Comment:
Does `<ENTER>` and/or `<SPACE>` get treated the same as a click? Does this logic tree work when interacting with a keyboard?

Reply:
Good call for additional testing. `<ENTER>` and `<SPACE>` are treated differently than clicks, but I have discovered that selecting an option with `<ENTER>` triggers both an `on-collapse` _and_ `on-expand` at the same time, which leads to some strange behavior. This fixes the problem, but it is very hacky and I would like to find a better solution. Do you have any suggestions?
```suggestion
            if (!event.wasClicked) setTimeout(() => this.el.firstChild.focus(), 0);
```

Reply:
Hmmm...very odd, indeed. This is a bit troubling. Before we try to fix this, especially in a hacky way, it might be worth delving into it a bit to get at the root cause.

> selecting an option with <ENTER> triggers both an on-collapse and on-expand at the same time, which leads to some strange behavior

That shouldn't be happening. I have a concern that this might be a bug in Marko itself. Maybe Marko is not capturing/trapping the keyboard event or double-firing it somehow? Dunno.  @ianmcburnie , @agliga , thoughts?